src/hooks/useButton.ts

```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react'
+
+export const useButton = () => {
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleClick = async <ExpectedResult>(callback: () => Promise<ExpectedResult | null>) => {
```

Comment:
it is safer to use `useCallback`. Otherwise, the hook that is dependent on this function will be triggered at every rendering
```suggestion
  const handleClick = useCallback(async <ExpectedResult>(callback: () => Promise<ExpectedResult | null>) => {
```

Reply:
> it is safer to use `useCallback`. Otherwise, the hook that is dependent on this function will be triggered at every rendering

Oh that is right updated here https://github.com/kixlab/kuiz/pull/84/commits/1d0f359ff29c7dd8523640bb26110dfa182d46e6 :)